src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java

```diff
@@ -1,34 +1,53 @@
 package frc.robot.hardware.rev.motors;
 
 import com.revrobotics.spark.config.SparkBaseConfig;
+import frc.robot.Robot;
 import frc.robot.hardware.ConnectedInputAutoLogged;
 import frc.robot.hardware.interfaces.IMotor;
 import frc.robot.hardware.interfaces.InputSignal;
+import frc.robot.hardware.mechanisms.MechanismSimulation;
+import frc.robot.hardware.rev.motors.simulation.SparkMaxSimulation;
 import frc.robot.hardware.signal.supplied.SuppliedAngleSignal;
 import frc.robot.hardware.signal.supplied.SuppliedDoubleSignal;
 import frc.utils.alerts.Alert;
 import frc.utils.alerts.AlertManager;
 import frc.utils.alerts.PeriodicAlert;
 import org.littletonrobotics.junction.Logger;
 
+import java.util.Optional;
+
 public abstract class SparkMaxMotor implements IMotor {
 
 	protected final SparkMaxWrapper motor;
+	private final Optional<SparkMaxSimulation> sparkMaxSimulationOptional;
 	private final String logPath;
 	private final ConnectedInputAutoLogged connectedInput;
 
-	public SparkMaxMotor(String logPath, SparkMaxWrapper motor) {
+	public SparkMaxMotor(String logPath, SparkMaxWrapper motor, MechanismSimulation mechanismSimulation) {
 		this.logPath = logPath;
 		this.motor = motor;
+		this.sparkMaxSimulationOptional = createSimulation(mechanismSimulation);
 
 		this.connectedInput = new ConnectedInputAutoLogged();
 		connectedInput.connected = true;
 
 		AlertManager.addAlert(new PeriodicAlert(Alert.AlertType.ERROR, logPath + "disconnectedAt", () -> !isConnected()));
 	}
 
+	public SparkMaxMotor(String logPath, SparkMaxWrapper motor) {
+		this(logPath, motor, null);
+	}
```

Comment:
are you sure null is the correct thing to insert? is there no default constructor?

Reply:
Same as the other comment